crates/bevy_math/src/rects/irect.rs

```diff
@@ -0,0 +1,429 @@
+        Self {
+            min: p0.min(p1),
+            max: p0.max(p1),
+        }
+    }
+
+    /// Create a new rectangle from its center and size.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if any of the components of the size is negative.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bevy_math::{IRect, IVec2};
+    /// let r = IRect::from_center_size(IVec2::ZERO, IVec2::splat(2)); // w=2 h=2
+    /// assert_eq!(r.min, IVec2::splat(-1));
+    /// assert_eq!(r.max, IVec2::splat(1));
+    /// ```
+    #[inline]
+    pub fn from_center_size(origin: IVec2, size: IVec2) -> Self {
+        debug_assert!(size.cmpge(IVec2::ZERO).all(), "IRect size must be positive");
+        let half_size = size / 2;
+        Self::from_center_half_size(origin, half_size)
+    }
+
+    /// Create a new rectangle from its center and half-size.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if any of the components of the half-size is negative.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bevy_math::{IRect, IVec2};
+    /// let r = IRect::from_center_half_size(IVec2::ZERO, IVec2::ONE); // w=2 h=2
+    /// assert_eq!(r.min, IVec2::splat(-1));
+    /// assert_eq!(r.max, IVec2::splat(1));
+    /// ```
+    #[inline]
+    pub fn from_center_half_size(origin: IVec2, half_size: IVec2) -> Self {
+        assert!(
+            half_size.cmpge(IVec2::ZERO).all(),
+            "IRect half_size must be positive"
+        );
+        Self {
+            min: origin - half_size,
+            max: origin + half_size,
+        }
+    }
+
+    /// Check if the rectangle is empty.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bevy_math::{IRect, IVec2};
+    /// let r = IRect::from_corners(IVec2::ZERO, IVec2::new(0, 1)); // w=0 h=1
+    /// assert!(r.is_empty());
+    /// ```
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.min.cmpge(self.max).any()
+    }
+
+    /// Rectangle width (max.x - min.x).
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bevy_math::IRect;
+    /// let r = IRect::new(0, 0, 5, 1); // w=5 h=1
+    /// assert_eq!(r.width(), 5);
+    /// ```
+    #[inline]
+    pub fn width(&self) -> i32 {
+        self.max.x - self.min.x
+    }
+
+    /// Rectangle height (max.y - min.y).
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bevy_math::IRect;
+    /// let r = IRect::new(0, 0, 5, 1); // w=5 h=1
+    /// assert_eq!(r.height(), 1);
+    /// ```
+    #[inline]
+    pub fn height(&self) -> i32 {
+        self.max.y - self.min.y
+    }
+
+    /// Rectangle size.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bevy_math::{IRect, IVec2};
+    /// let r = IRect::new(0, 0, 5, 1); // w=5 h=1
+    /// assert_eq!(r.size(), IVec2::new(5, 1));
+    /// ```
+    #[inline]
+    pub fn size(&self) -> IVec2 {
+        self.max - self.min
+    }
+
+    /// Rectangle half-size.
```

Comment:
Need to call out the rounding behavior here since we deal with halved integral numbers.

Reply:
Thanks for the change. Could you please also change the example to use one odd number, for example the width? That way it's representative of that rounding behavior.